proposals/lip-0053.md

```diff
@@ -215,8 +251,10 @@ def verifyPartnerChainOutboxRoot(ccu: Transaction) -> None:
     newInboxRoot = calculateRootFromRightWitness(inboxTree.size, inboxTree.appendPath, ccu.params.inboxUpdate.messageWitnessHashes)
 
     # Verification for non-empty certificates.
+    outboxRootWitness = ccu.params.inboxUpdate.outboxRootWitness
     if len(ccu.params.certificate) > 0:
-        outboxRootWitness = ccu.params.inboxUpdate.outboxRootWitness
+        if outboxRootWitness is empty:
+            raise Exception("If a certificate is present, the outbox root witness must be non-empty to authenticate the new partnerChainOutboxRoot.")
```

Comment:
I would remove the "if a certificate is present" from the sentence, because it's confusing.

In general, you might have a certificate and an empty outboxRootWitness, in case inbox update is empty. But here this function is called only when inbox update is non-empty. To avoid confusion, I would just say, "the outbox root must be non-empty to .... "